src/shared/bus-util.h

```diff
@@ -39,22 +39,8 @@ int bus_connect_user_systemd(sd_bus **_bus);
 int bus_connect_transport(BusTransport transport, const char *host, bool user, sd_bus **bus);
 int bus_connect_transport_systemd(BusTransport transport, const char *host, bool user, sd_bus **bus);
 
-#define bus_log_address_error(r)                                        \
-        ({                                                              \
-                int _k = (r);                                           \
-                log_error_errno(_k,                                     \
-                                _k == -ENOMEDIUM ? "Failed to set bus address: $DBUS_SESSION_BUS_ADDRESS and $XDG_RUNTIME_DIR not defined (consider using --machine=<user>@.host --user to connect to bus of other user)" : \
-                                                   "Failed to set bus address: %m"); \
-        })
-
-#define bus_log_connect_error(r)                                        \
-        ({                                                              \
-                int _k = (r);                                           \
-                log_error_errno(_k,                                     \
-                                _k == -ENOMEDIUM       ? "Failed to connect to bus: $DBUS_SESSION_BUS_ADDRESS and $XDG_RUNTIME_DIR not defined (consider using --machine=<user>@.host --user to connect to bus of other user)" : \
-                                ERRNO_IS_PRIVILEGE(_k) ? "Failed to connect to bus: Operation not permitted (consider using --machine=<user>@.host --user to connect to bus of other user)" : \
-                                                         "Failed to connect to bus: %m"); \
-        })
+int bus_log_address_error(int r);
+int bus_log_connect_error(int r);
```

Comment:
I don't want to start a bikeshed, but I assume you considered making these `static inline` and it would have no benefit?

Reply:
I considered it, but since the second commit adds quite a bit of logic, it didn't seem worthwhile. Those functions are called dozens of times, so with inline there'd be a lot of duplication.

Reply:
yeah sounds reasonable, just wanted to check